### PR TITLE
refactor: Change toasts validation to use OnAccessibilityEventListener

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
https://github.com/appium/appium-espresso-driver/issues/617

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetToastVisibility.kt [hunk 1]
```diff
@@ -16,35 +16,25 @@
...
-            }
-            throw e
+        val toastMessage = NotificationListener.toastMessage
+        if (toastMessage.isEmpty()) {
+            return false
+        }
+        return if (params.isRegexp) {
+            Pattern.compile(params.text)
+                    .matcher(toastMessage.joinToString(separator = "\n"))
+                    .matches()
```
Comment (KazuCocoa) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetToastVisibility.kt:34]: `.matches()` should be `.find()` as io.appium.espressoserver.lib.viewmatcher.withRegexp
(The result could be different)
Or `params.text.toRegex().containsMatchIn(toastMessage.joinToString(separator = "\n"))` like Kotlin? :)

Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetToastVisibility.kt]: changed

**Review by KazuCocoa — APPROVED** [review]

**Comment by KazuCocoa** [comment]
Mmm... > mocha

**Comment by mykola-mokhnach** [comment]
> Mmm... > mocha

Yeah, I gave up on it :(

**Comment by KazuCocoa** [comment]
I ran some tests in ruby_lib_core with this module, then no regression occurred.
We probably can merge this (reverting mocha stuff) for now and create a version to fix the current toast behavior.

**Comment by mykola-mokhnach** [comment]
yes, locally all works perfectly for me, but CI miserably fails :(

**Pull request merged**